src/nat/data_models/optimizable.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Sequence
+from typing import Any
+from typing import Generic
+from typing import TypeVar
+
+from optuna import Trial
+from pydantic import BaseModel
+from pydantic import Field
+
+T = TypeVar("T", int, float, bool, str)
+
+
+# --------------------------------------------------------------------- #
+# 1.  Hyper‑parameter metadata container                                #
+# --------------------------------------------------------------------- #
+class SearchSpace(BaseModel, Generic[T]):
+    low: T | Sequence[T] | None = None
+    high: T | None = None
```

Comment:
Looking at how discrete options are specified, perhaps it would be better to have an explicit "values" field?

Then we could do an "after" model validate to ensure mutual exclusion between the three categories of search space:
- `values`
- `is_prompt` + `prompt`
- `high` + `low` + `step` (opt) + `log` (opt)
- everything is `None` (Sentinel)
```suggestion
class SearchSpace(BaseModel, Generic[T]):
    values: Sequence[T] | None = None
    low: T | None = None
    high: T | None = None
```